### PR TITLE
ci: Migrate to ViteJS as the storybook bundler

### PR DESCRIPTION
**Pull request by dherault**
This PR adds ViteJS as the storybook bundler. Hence storybook loads faster :rocket:!

### CODE REVIEW

**Comment by DavidSpek** [comment]
Error: No release type found in pull request title "Migrate to ViteJS as the storybook bundler". Add a prefix to indicate what kind of release this pull request corresponds to. For reference, see https://www.conventionalcommits.org/

Available types:
 - feat: A new feature
 - fix: A bug fix
 - docs: Documentation only changes
 - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
 - refactor: A code change that neither fixes a bug nor adds a feature
 - perf: A code change that improves performance
 - test: Adding missing tests or correcting existing tests
 - build: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
 - ci: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
 - chore: Other changes that don't modify src or test files
 - revert: Reverts a previous commit

File: package.json [hunk 1]
```diff
@@ -61,18 +61,17 @@
...
-    "@pluralsh/eslint-config-typescript": "2.5.3",
-    "@storybook/addon-actions": "6.5.13",
-    "@storybook/addon-docs": "6.5.13",
-    "@storybook/addon-essentials": "6.5.13",
-    "@storybook/addon-interactions": "6.5.13",
-    "@storybook/addon-links": "6.5.13",
-    "@storybook/builder-webpack5": "6.5.13",
-    "@storybook/manager-webpack5": "6.5.13",
-    "@storybook/node-logger": "6.5.13",
+    "@pluralsh/eslint-config-typescript": "^2.5.3",
```
Comment (floreks) [inline package.json:64]: nit: use exact version match

File: package.json [hunk 2]
```diff
@@ -83,24 +82,30 @@
     "fuse.js": "6.6.2",
     "honorable": "0.148.0",
     "honorable-theme-default": "0.66.0",
+    "jest-mock": "^29.3.1",
```
Comment (floreks) [inline package.json:85]: same here

File: package.json [hunk 3]
```diff
@@ -83,24 +82,30 @@
...
     "react-dom": "18.2.0",
     "react-router-dom": "6.4.3",
     "react-scripts": "5.0.1",
     "react-transition-group": "4.4.5",
     "rimraf": "3.0.2",
+    "storybook": "7.0.0-alpha.49",
     "styled-components": "5.3.6",
-    "typescript": "4.9.3"
+    "typescript": "4.9.3",
+    "vite": "^3.2.4"
```
Comment (floreks) [inline package.json:95]: and here :slightly_smiling_face:

**Review by floreks — APPROVED** [review]

**Pull request merged**